cvat-data/src/js/cvat-data.js

```diff
@@ -141,96 +58,49 @@ class FrameProvider {
 
     */
 
-    startDecode(block, start, end, callback) {
-        if (this._running) {
-            const error = new Error('Decoding has already running');
-            const promise = new Promise((resolve, reject) => {
-                if (this._promise) {
-                    this._promise.reject();
-                }
-
-                this._promise = {
-                    resolve,
-                    reject,
-                };
-            });
-            error.donePromise = promise;
-            throw error;
-        }
+    startDecode(block, start, end, callback)
+    {
+         if (this._blockType === BlockType.TSVIDEO){
+            if (this._running) {
+                throw new Error('Decoding has already running');
+            }
 
-        this._running = true;
-        this._blocks.push('-1:-1');
+            this._blocks.push(`${start}:${end}`);
+            this._cleanup();            
 
-        if (this._blockType === BlockType.TSVIDEO) {
-            this._demuxer.write(block);
-            this._cleanup();
-            this._currFrame = start;
-            for (let i = start; i <= end; i++) {
-                this._frames[i] = 'loading';
-            }
-            this._decode_id = setTimeout(
-                this.decode.bind(this, start, end, callback),
-                10,
-            );
-        } else {
-            const zip = new JSZip();
-            zip.loadAsync(block).then((_zip) => {
-                let index = start;
-                _zip.forEach((relativePath) => {
-                    const fileIndex = index++;
-                    _zip.file(relativePath).async('blob').then((fileData) => {
-                        // Need to be sure that files
-                        // are extracted one by one
-                        // and don't break an order
-                        const reader = new FileReader();
-                        reader.onload = () => {
-                            this._frames[fileIndex] = reader.result;
-                            this._blocks.pop().split(':');
-                            this._blocks.push(`${start}:${fileIndex}`);
-                            callback(fileIndex);
-                            if (fileIndex > end) {
-                                this._doneDecode();
-                            }
-                        };
-                        reader.readAsDataURL(fileData);
-                    });
-                });
+            const worker = new Worker('decode_video.js');
+            worker.postMessage({block : block, 
+                                start : start, 
+                                  end : end });
+            
+            worker.onerror(function (event) {
+                console.log(['ERROR: Line ', e.lineno, ' in ', e.filename, ': ', e.message].join(''));
```

Comment:
I suppose console.log is not enough in this case
We should think about error handling together